kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/DeltaTableSchemaEvolutionSuite.scala

```diff
@@ -978,7 +978,12 @@ class DeltaTableSchemaEvolutionSuite extends DeltaTableWriteSuiteBase with Colum
               true),
             false),
           true,
-          fieldMetadataForMapColumn(4, "map", "map", 5, 6))
```

Comment:
This test case was testing the wrong thing before and the change not allowing lower fieldIds highlighted it. We were not using the fieldIds here that were used in the original schema and the entire "map" column was being treated as a new column.